### ISSUE TITLE
kube-apiserver : /healthz returns 401 Unauthorized

### ISSUE THREAD

**Issue by ronny35** [rancher/rke#557]
**RKE version:**
v0.1.6-rc3

**Docker version: (`docker version`,`docker info` preferred)**
17.03.2-ce

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
debian

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
Bare-metal

**cluster.yml file:**
```
nodes:
- address: 192.168.56.102
  port: "22"
  internal_address: 192.168.56.102
  role:
  - controlplane
  - etcd
  hostname_override: ""
  user: user
  docker_socket: /var/run/docker.sock
  ssh_key: ""
  ssh_key_path: ~/.ssh/id_rsa
  labels: {}
- address: 192.168.56.101
  port: "22"
  internal_address: 192.168.56.101
  role:
  - worker
  hostname_override: ""
  user: user
  docker_socket: /var/run/docker.sock
  ssh_key: ""
  ssh_key_path: ~/.ssh/id_rsa
  labels: {}
- address: 192.168.56.103
  port: "22"
  internal_address: 192.168.56.103
  role:
  - worker
  hostname_override: ""
  user: user
  docker_socket: /var/run/docker.sock
  ssh_key: ""
  ssh_key_path: ~/.ssh/id_rsa
  labels: {}
services:
  etcd:
    image: rancher/coreos-etcd:v3.1.12
    extra_args: {}
    extra_binds: []
    external_urls: []
    ca_cert: ""
    cert: ""
    key: ""
    path: ""
  kube-api:
    image: rancher/hyperkube:v1.10.1-rancher1
    extra_args: {}
    extra_binds: []
    service_cluster_ip_range: 10.43.0.0/16
    pod_security_policy: false
  kube-controller:
    image: rancher/hyperkube:v1.10.1-rancher1
    extra_args: {}
    extra_binds: []
    cluster_cidr: 10.42.0.0/16
    service_cluster_ip_range: 10.43.0.0/16
  scheduler:
    image: rancher/hyperkube:v1.10.1-rancher1
    extra_args: {}
    extra_binds: []
  kubelet:
    image: rancher/hyperkube:v1.10.1-rancher1
    extra_args: {}
    extra_binds: []
    cluster_domain: cluster.local
    infra_container_image: rancher/pause-amd64:3.1
    cluster_dns_server: 10.43.0.10
    fail_swap_on: false
  kubeproxy:
    image: rancher/hyperkube:v1.10.1-rancher1
    extra_args: {}
    extra_binds: []
network:
  plugin: canal
  options: {}
authentication:
  strategy: x509
  options: {}
  sans: []
addons: ""
addons_include: []
system_images:
  etcd: ""
  alpine: ""
  nginx_proxy: ""
  cert_downloader: ""
  kubernetes_services_sidecar: ""
  kubedns: ""
  dnsmasq: ""
  kubedns_sidecar: ""
  kubedns_autoscaler: ""
  kubernetes: ""
  flannel: ""
  flannel_cni: ""
  calico_node: ""
  calico_cni: ""
  calico_controllers: ""
  calico_ctl: ""
  canal_node: ""
  canal_cni: ""
  canal_flannel: ""
  wave_node: ""
  weave_cni: ""
  pod_infra_container: ""
  ingress: ""
  ingress_backend: ""
ssh_key_path: ~/.ssh/id_rsa
ssh_agent_auth: false
authorization:
  mode: none
  options: {}
ignore_docker_version: false
kubernetes_version: ""
private_registries: []
ingress:
  provider: ""
  options: {}
  node_selector: {}
  extra_args: {}
cluster_name: ""
cloud_provider:
  name: ""
  cloud_config: {}
prefix_path: ""
```

**Steps to Reproduce:**
./rke up --config cluster.yml

**Results:**
```
FATA[0066] [controlPlane] Failed to bring up Control Plane: Failed to verify healthcheck: Service [kube-apiserver] is not healthy on host [192.168.56.102]. Response code: [401], response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"Unauthorized","reason":"Unauthorized","code":401}
```

**Comment by superseb** [comment]
Can you supply the full output of `docker logs --tail=all etcd` and `docker logs --tail=all kube-apiserver`?

**Comment by ronny35** [comment]
**etcd.logs**

2018-04-28 17:24:35.542547 I | etcdmain: etcd Version: 3.1.12
2018-04-28 17:24:35.542621 I | etcdmain: Git SHA: 918698add
2018-04-28 17:24:35.542638 I | etcdmain: Go Version: go1.8.7
2018-04-28 17:24:35.542654 I | etcdmain: Go OS/Arch: linux/amd64
2018-04-28 17:24:35.542668 I | etcdmain: setting maximum number of CPUs to 1, total number of available CPUs is 1
2018-04-28 17:24:35.542681 N | etcdmain: failed to detect default host (could not find default route)
2018-04-28 17:24:35.542727 I | embed: peerTLS: cert = /etc/kubernetes/ssl/kube-etcd-192-168-56-102.pem, key = /etc/kubernetes/ssl/kube-etcd-192-168-56-102-key.pem, ca = , trusted-ca = /etc/kubernetes/ssl/kube-ca.pem, client-cert-auth = true
2018-04-28 17:24:35.548872 I | embed: listening for peers on https://0.0.0.0:2380
2018-04-28 17:24:35.549143 I | embed: listening for client requests on 0.0.0.0:2379
2018-04-28 17:24:35.560276 I | warning: ignoring ServerName for user-provided CA for backwards compatibility is deprecated
2018-04-28 17:24:35.560338 I | etcdserver: name = etcd-192.168.56.102
2018-04-28 17:24:35.560344 I | etcdserver: data dir = /var/lib/rancher/etcd
2018-04-28 17:24:35.560349 I | etcdserver: member dir = /var/lib/rancher/etcd/member
2018-04-28 17:24:35.560352 I | etcdserver: heartbeat = 100ms
2018-04-28 17:24:35.560355 I | etcdserver: election = 1000ms
2018-04-28 17:24:35.560359 I | etcdserver: snapshot count = 10000
2018-04-28 17:24:35.560365 I | etcdserver: advertise client URLs = https://192.168.56.102:2379,https://192.168.56.102:4001
2018-04-28 17:24:35.560369 I | etcdserver: initial advertise peer URLs = https://192.168.56.102:2380
2018-04-28 17:24:35.560374 I | etcdserver: initial cluster = etcd-192.168.56.102=https://192.168.56.102:2380
2018-04-28 17:24:35.563566 I | etcdserver: starting member 9f71024655e5ae6c in cluster 7db71c13ad9ce6c7
2018-04-28 17:24:35.563584 I | raft: 9f71024655e5ae6c became follower at term 0
2018-04-28 17:24:35.563593 I | raft: newRaft 9f71024655e5ae6c [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
2018-04-28 17:24:35.563598 I | raft: 9f71024655e5ae6c became follower at term 1
2018-04-28 17:24:35.588862 I | warning: ignoring ServerName for user-provided CA for backwards compatibility is deprecated
2018-04-28 17:24:35.589241 I | warning: ignoring ServerName for user-provided CA for backwards compatibility is deprecated
2018-04-28 17:24:35.589275 I | etcdserver: starting server... [version: 3.1.12, cluster version: to_be_decided]
2018-04-28 17:24:35.589306 I | embed: ClientTLS: cert = /etc/kubernetes/ssl/kube-etcd-192-168-56-102.pem, key = /etc/kubernetes/ssl/kube-etcd-192-168-56-102-key.pem, ca = , trusted-ca = /etc/kubernetes/ssl/kube-ca.pem, client-cert-auth = true
2018-04-28 17:24:35.590392 I | etcdserver/membership: added member 9f71024655e5ae6c [https://192.168.56.102:2380] to cluster 7db71c13ad9ce6c7
2018-04-28 17:24:36.364124 I | raft: 9f71024655e5ae6c is starting a new election at term 1
2018-04-28 17:24:36.364228 I | raft: 9f71024655e5ae6c became candidate at term 2
2018-04-28 17:24:36.364251 I | raft: 9f71024655e5ae6c received MsgVoteResp from 9f71024655e5ae6c at term 2
2018-04-28 17:24:36.364271 I | raft: 9f71024655e5ae6c became leader at term 2
2018-04-28 17:24:36.364287 I | raft: raft.node: 9f71024655e5ae6c elected leader 9f71024655e5ae6c at term 2
2018-04-28 17:24:36.364691 I | etcdserver: setting up the initial cluster version to 3.1
2018-04-28 17:24:36.368783 N | etcdserver/membership: set the initial cluster version to 3.1
2018-04-28 17:24:36.368830 I | etcdserver/api: enabled capabilities for version 3.1
2018-04-28 17:24:36.368857 I | etcdserver: published {Name:etcd-192.168.56.102 ClientURLs:[https://192.168.56.102:2379 https://192.168.56.102:4001]} to cluster 7db71c13ad9ce6c7
2018-04-28 17:24:36.368959 I | embed: ready to serve client requests
2018-04-28 17:24:36.369148 I | embed: serving client requests on [::]:2379
2018-04-28 17:34:36.669177 I | mvcc: store.index: compact 34
2018-04-28 17:34:36.669748 I | mvcc: finished scheduled compaction at 34 (took 75.527µs)
2018-04-28 17:39:36.679221 I | mvcc: store.index: compact 35
2018-04-28 17:39:36.680274 I | mvcc: finished scheduled compaction at 35 (took 59.766µs)
2018-04-28 17:44:36.692583 I | mvcc: store.index: compact 36
2018-04-28 17:44:36.693211 I | mvcc: finished scheduled compaction at 36 (took 60.132µs)
2018-04-28 17:49:36.704666 I | mvcc: store.index: compact 37
2018-04-28 17:49:36.705909 I | mvcc: finished scheduled compaction at 37 (took 60.846µs)
2018-04-28 17:54:36.720299 I | mvcc: store.index: compact 38
2018-04-28 17:54:36.720972 I | mvcc: finished scheduled compaction at 38 (took 60.477µs)
2018-04-28 17:59:36.728678 I | mvcc: store.index: compact 39
2018-04-28 17:59:36.728856 I | mvcc: finished scheduled compaction at 39 (took 17.248µs)
2018-04-28 18:04:36.735794 I | mvcc: store.index: compact 40
2018-04-28 18:04:36.736068 I | mvcc: finished scheduled compaction at 40 (took 17.448µs)
2018-04-28 18:09:36.745430 I | mvcc: store.index: compact 41
2018-04-28 18:09:36.746047 I | mvcc: finished scheduled compaction at 41 (took 90.197µs)
2018-04-28 18:14:36.751933 I | mvcc: store.index: compact 42
2018-04-28 18:14:36.752160 I | mvcc: finished scheduled compaction at 42 (took 18.971µs)
2018-04-28 18:19:36.758214 I | mvcc: store.index: compact 43
2018-04-28 18:19:36.758543 I | mvcc: finished scheduled compaction at 43 (took 33.289µs)
2018-04-28 18:24:36.766232 I | mvcc: store.index: compact 44
2018-04-28 18:24:36.766575 I | mvcc: finished scheduled compaction at 44 (took 31.135µs)
2018-04-28 18:29:36.777538 I | mvcc: store.index: compact 45
2018-04-28 18:29:36.778380 I | mvcc: finished scheduled compaction at 45 (took 59.186µs)
2018-04-28 18:34:36.790054 I | mvcc: store.index: compact 46
2018-04-28 18:34:36.791095 I | mvcc: finished scheduled compaction at 46 (took 59.365µs)
2018-04-28 18:39:36.801960 I | mvcc: store.index: compact 47
2018-04-28 18:39:36.802589 I | mvcc: finished scheduled compaction at 47 (took 64.319µs)
2018-04-28 18:44:36.813747 I | mvcc: store.index: compact 48
2018-04-28 18:44:36.814667 I | mvcc: finished scheduled compaction at 48 (took 88.905µs)
2018-04-28 18:49:36.823984 I | mvcc: store.index: compact 49
2018-04-28 18:49:36.824392 I | mvcc: finished scheduled compaction at 49 (took 42.386µs)
2018-04-28 18:54:36.833624 I | mvcc: store.index: compact 50
2018-04-28 18:54:36.834617 I | mvcc: finished scheduled compaction at 50 (took 57.51µs)
2018-04-28 18:59:36.842175 I | mvcc: store.index: compact 51
2018-04-28 18:59:36.842389 I | mvcc: finished scheduled compaction at 51 (took 19.673µs)
2018-04-28 19:04:36.848461 I | mvcc: store.index: compact 52
2018-04-28 19:04:36.848653 I | mvcc: finished scheduled compaction at 52 (took 17.393µs)
2018-04-28 19:09:36.855347 I | mvcc: store.index: compact 53
2018-04-28 19:09:36.855720 I | mvcc: finished scheduled compaction at 53 (took 33.533µs)
2018-04-28 19:14:36.862363 I | mvcc: store.index: compact 54
2018-04-28 19:14:36.862541 I | mvcc: finished scheduled compaction at 54 (took 18.216µs)
2018-04-28 19:19:36.870183 I | mvcc: store.index: compact 55
2018-04-28 19:19:36.870414 I | mvcc: finished scheduled compaction at 55 (took 19.582µs)
2018-04-28 19:24:36.879148 I | mvcc: store.index: compact 56
2018-04-28 19:24:36.879558 I | mvcc: finished scheduled compaction at 56 (took 34.933µs)
2018-04-28 19:29:36.887279 I | mvcc: store.index: compact 57
2018-04-28 19:29:36.887562 I | mvcc: finished scheduled compaction at 57 (took 26.313µs)
2018-04-28 19:34:36.894063 I | mvcc: store.index: compact 58
2018-04-28 19:34:36.894371 I | mvcc: finished scheduled compaction at 58 (took 27.358µs)
2018-04-28 19:39:36.900502 I | mvcc: store.index: compact 59
2018-04-28 19:39:36.900769 I | mvcc: finished scheduled compaction at 59 (took 23.497µs)

**kube-apiserver**
+ grep -q cloud-provider=azure
+ echo kube-apiserver --kubelet-client-key=/etc/kubernetes/ssl/kube-apiserver-key.pem --allow-privileged=true --storage-backend=etcd3 --cloud-provider= --service-cluster-ip-range=10.43.0.0/16 --tls-private-key-file=/etc/kubernetes/ssl/kube-apiserver-key.pem --secure-port=6443 --kubelet-preferred-address-types=InternalIP,ExternalIP,Hostname --client-ca-file=/etc/kubernetes/ssl/kube-ca.pem --tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305 --insecure-port=0 --tls-cert-file=/etc/kubernetes/ssl/kube-apiserver.pem --kubelet-client-certificate=/etc/kubernetes/ssl/kube-apiserver.pem --insecure-bind-address=127.0.0.1 --bind-address=0.0.0.0 --admission-control=ServiceAccount,NamespaceLifecycle,LimitRanger,PersistentVolumeLabel,DefaultStorageClass,ResourceQuota,DefaultTolerationSeconds --service-account-key-file=/etc/kubernetes/ssl/kube-apiserver-key.pem --advertise-address=192.168.56.102 --etcd-cafile=/etc/kubernetes/ssl/kube-ca.pem --etcd-certfile=/etc/kubernetes/ssl/kube-node.pem --etcd-keyfile=/etc/kubernetes/ssl/kube-node-key.pem --etcd-servers=https://192.168.56.102:2379 --etcd-prefix=/registry
+ '[' kube-apiserver = kubelet ']'
+ exec kube-apiserver --kubelet-client-key=/etc/kubernetes/ssl/kube-apiserver-key.pem --allow-privileged=true --storage-backend=etcd3 --cloud-provider= --service-cluster-ip-range=10.43.0.0/16 --tls-private-key-file=/etc/kubernetes/ssl/kube-apiserver-key.pem --secure-port=6443 --kubelet-preferred-address-types=InternalIP,ExternalIP,Hostname --client-ca-file=/etc/kubernetes/ssl/kube-ca.pem --tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305 --insecure-port=0 --tls-cert-file=/etc/kubernetes/ssl/kube-apiserver.pem --kubelet-client-certificate=/etc/kubernetes/ssl/kube-apiserver.pem --insecure-bind-address=127.0.0.1 --bind-address=0.0.0.0 --admission-control=ServiceAccount,NamespaceLifecycle,LimitRanger,PersistentVolumeLabel,DefaultStorageClass,ResourceQuota,DefaultTolerationSeconds --service-account-key-file=/etc/kubernetes/ssl/kube-apiserver-key.pem --advertise-address=192.168.56.102 --etcd-cafile=/etc/kubernetes/ssl/kube-ca.pem --etcd-certfile=/etc/kubernetes/ssl/kube-node.pem --etcd-keyfile=/etc/kubernetes/ssl/kube-node-key.pem --etcd-servers=https://192.168.56.102:2379 --etcd-prefix=/registry
Flag --insecure-port has been deprecated, This flag will be removed in a future version.
Flag --insecure-bind-address has been deprecated, This flag will be removed in a future version.
Flag --admission-control has been deprecated, Use --enable-admission-plugins or --disable-admission-plugins instead. Will be removed in a future version.
I0428 17:24:36.283465       1 server.go:135] Version: v1.10.1
I0428 17:24:36.283518       1 server.go:717] external host was not specified, using 192.168.56.102
W0428 17:24:36.283539       1 authentication.go:377] AnonymousAuth is not allowed with the AllowAll authorizer.  Resetting AnonymousAuth to false. You should use a different authorizer
W0428 17:24:36.641519       1 admission.go:68] PersistentVolumeLabel admission controller is deprecated. Please remove this controller from your configuration files and scripts.
I0428 17:24:36.645028       1 plugins.go:149] Loaded 7 admission controller(s) successfully in the following order: NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultTolerationSeconds,PersistentVolumeLabel,DefaultStorageClass,ResourceQuota.
W0428 17:24:36.645435       1 admission.go:68] PersistentVolumeLabel admission controller is deprecated. Please remove this controller from your configuration files and scripts.
I0428 17:24:36.645566       1 plugins.go:149] Loaded 7 admission controller(s) successfully in the following order: NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultTolerationSeconds,PersistentVolumeLabel,DefaultStorageClass,ResourceQuota.
I0428 17:24:36.648103       1 master.go:228] Using reconciler: master-count
W0428 17:24:36.995571       1 genericapiserver.go:342] Skipping API batch/v2alpha1 because it has no resources.
W0428 17:24:37.003106       1 genericapiserver.go:342] Skipping API rbac.authorization.k8s.io/v1alpha1 because it has no resources.
W0428 17:24:37.004152       1 genericapiserver.go:342] Skipping API storage.k8s.io/v1alpha1 because it has no resources.
W0428 17:24:37.092738       1 genericapiserver.go:342] Skipping API admissionregistration.k8s.io/v1alpha1 because it has no resources.
[restful] 2018/04/28 17:24:37 log.go:33: [restful/swagger] listing is available at https://192.168.56.102:6443/swaggerapi
[restful] 2018/04/28 17:24:37 log.go:33: [restful/swagger] https://192.168.56.102:6443/swaggerui/ is mapped to folder /swagger-ui/
[restful] 2018/04/28 17:24:38 log.go:33: [restful/swagger] listing is available at https://192.168.56.102:6443/swaggerapi
[restful] 2018/04/28 17:24:38 log.go:33: [restful/swagger] https://192.168.56.102:6443/swaggerui/ is mapped to folder /swagger-ui/
W0428 17:24:38.567340       1 admission.go:68] PersistentVolumeLabel admission controller is deprecated. Please remove this controller from your configuration files and scripts.
I0428 17:24:38.567637       1 plugins.go:149] Loaded 7 admission controller(s) successfully in the following order: NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultTolerationSeconds,PersistentVolumeLabel,DefaultStorageClass,ResourceQuota.
I0428 17:24:41.146246       1 serve.go:96] Serving securely on [::]:6443
I0428 17:24:41.146332       1 controller.go:84] Starting OpenAPI AggregationController
I0428 17:24:41.148086       1 crd_finalizer.go:242] Starting CRDFinalizer
I0428 17:24:41.148765       1 apiservice_controller.go:90] Starting APIServiceRegistrationController
I0428 17:24:41.148789       1 cache.go:32] Waiting for caches to sync for APIServiceRegistrationController controller
I0428 17:24:41.148829       1 available_controller.go:262] Starting AvailableConditionController
I0428 17:24:41.148844       1 cache.go:32] Waiting for caches to sync for AvailableConditionController controller
I0428 17:24:41.150134       1 customresource_discovery_controller.go:174] Starting DiscoveryController
I0428 17:24:41.150181       1 naming_controller.go:276] Starting NamingConditionController
I0428 17:24:41.150838       1 crdregistration_controller.go:110] Starting crd-autoregister controller
I0428 17:24:41.150859       1 controller_utils.go:1019] Waiting for caches to sync for crd-autoregister controller
I0428 17:24:41.207956       1 controller.go:537] quota admission added evaluator for: { endpoints}
I0428 17:24:41.252917       1 controller_utils.go:1026] Caches are synced for crd-autoregister controller
I0428 17:24:41.252948       1 cache.go:39] Caches are synced for APIServiceRegistrationController controller
I0428 17:24:41.252958       1 cache.go:39] Caches are synced for AvailableConditionController controller
I0428 17:24:41.252978       1 autoregister_controller.go:136] Starting autoregister controller
I0428 17:24:41.252984       1 cache.go:32] Waiting for caches to sync for autoregister controller
I0428 17:24:41.353155       1 cache.go:39] Caches are synced for autoregister controller

**Comment by superseb** [comment]
Seems to be a duplicate of https://github.com/rancher/rke/issues/235, can you run it with RBAC enabled for now and see if that solves it. If not, please report back.
